### PR TITLE
Enable Upstream Testing

### PR DESCRIPTION
**Pull request by bryanculver**
- Updated Dockerfile to be compatible with upstream testing
- Added workflow fragment to trigger testing
- Updated CI to run 3.10 unittests
- Bumped Nautobot minimal version to 1.4
- Fixed tests failing with Nautobot 1.5.12
- Disabled tests that can't be fixed, blocked by https://github.com/nautobot/nautobot/issues/3419

### CODE REVIEW

File: development/Dockerfile
Comment (bryanculver) [inline development/Dockerfile:61]: This deviates from the standard because this plugin publishes extras.

`--extras metrics` is what has been added

File: .github/workflows/upstream_testing.yml
Comment (bryanculver) [inline .github/workflows/upstream_testing.yml:8]:
```suggestion
```
Remove before merging.

**Comment by bryanculver** [comment]
`next` is expected to fail because it goes to a new major version of Nautobot so can be safely ignored for now.

File: development/Dockerfile
Comment (bryanculver) [inline development/Dockerfile:13]: There is a mismatch here (and in the previous version of this file):

**Nautobot Version**
- `1.4.7` previous default in Dockerfile
- `latest` current invoke default: https://github.com/nautobot/nautobot-plugin-circuit-maintenance/blob/5bbf9c8763b9f6e255703ca269ba95b2c63df32c/tasks.py#L43
- `1.0.0-beta.4` current minimum in `PluginConfig`: https://github.com/nautobot/nautobot-plugin-circuit-maintenance/blob/5bbf9c8763b9f6e255703ca269ba95b2c63df32c/nautobot_circuit_maintenance/__init__.py#L86 
- `*` current minimum in `pyproject.toml`. https://github.com/nautobot/nautobot-plugin-circuit-maintenance/blob/5bbf9c8763b9f6e255703ca269ba95b2c63df32c/pyproject.toml#L23

**Python Version**
- `3.8` previous default in Dockerfile
- `3.7` current defaults in invoke and `pyproject.toml`

I will go ahead and update these in my PR but I would like to know the maintainer's preferences.

Comment (chadell) [reply development/Dockerfile:13]: I would use Nautobot latest for development, while leaving the all Nautobot versions supported. For Python, I would go with 3.8 for development, and keep testing for 3.7 in the pipeline.
is this in line with your recommendations?

File: development/Dockerfile
Comment (chadell) [inline development/Dockerfile:56]: why not using poetry 1.2 then?

File: development/Dockerfile
Comment (chadell) [reply development/Dockerfile:61]: to better better understand. Why we don't use poetry to install instead of translating to requirements and install with Pip? (maybe it's somewhere explained by I missed it)

File: development/Dockerfile
Comment (glennmatthews) [reply development/Dockerfile:56]: Older Nautobot-dev Docker images come with poetry 1.1.x. IIRC we'd have to explicitly upgrade poetry to latest as part of this Dockerfile, though that might be desirable in any case?

Comment (bryanculver) [reply development/Dockerfile:56]: Glenn's point is correct. Is this blocking getting started with automated testing?

Comment (chadell) [reply development/Dockerfile:56]: nope, just informational question

File: development/Dockerfile
Comment (chadell) [reply development/Dockerfile:61]: ok, understood

**Comment by chadell** [comment]
to refresh, @bryanculver what is missing in this PR to move it forward? how could we help you?

**Comment by bryanculver** [comment]
> to refresh, @bryanculver what is missing in this PR to move it forward? how could we help you?

Making updates similar to those done in Chatops: https://github.com/nautobot/nautobot-plugin-chatops/pull/171

Namely:

- Bring up the minimum Nautobot version to 1.4+
- Update CI to ensure we're testing on the right versions
- Update to the latest example of the upstream_testing.yml and Dockerfiles (see cookie-cutter or Chatops PR for the example)
- Addressing conflicts like this: https://github.com/nautobot/nautobot-plugin-circuit-maintenance/pull/228#issuecomment-1339813292
    - These plugins are never intended to be installed standalone and should be running on dev base images so you should get dependencies like this for free

**Comment by snaselj** [comment]
Rebased on top of `develop`

**Comment by snaselj** [comment]
Rebased on top of `develop`

**Comment by snaselj** [comment]
The last test failing is `upstream-test` on `next`

**Comment by snaselj** [comment]
This PR is ready for review now.

**Review by chadell — APPROVED** [review]
🚀 LGTM!

**Pull request merged**
